CONTRIBUTING.md

```diff
@@ -15,6 +15,58 @@ You generally only need to submit a CLA once, so if you've already submitted one
 (even if it was for a different project), you probably don't need to do it
 again.
 
+## Tests
+
+### Unit Tests
+
+Unit tests can be run locally with minimal fuss.
+
+**Requirements**
+
+* [Node.js](https://nodejs.org/en/download/).
+
+**Running Unit Tests**
+
+```sh
+bazel test //...
+```
+
+### End-to-end Tests
+
+End-to-end tests require the provisioning of external services.
+
+**Requirements**
+
+* A container registry (e.g. [Google Container Registry](https://cloud.google.com/container-registry))
+* A Kubernetes cluster (e.g. [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine))
+* The host you are running tests from must match the architecture of the Kubernetes nodes (i.e. you cannot run tests on MacOS if your Kubernetes nodes run Linux)
```

Comment:
This is only partially true, for Go you can add a flag to `.bazelrc` that will make everything cross-compile to `linux/amd64` and things work fine.  For Python/Node this is true :-/